portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+/*
+ * Counts the number of times each city occurs in a list of markers. Returns {[city, count],...}
+ */
+function countMarkers(markers) {
+  const markerCounts = {};
+  for(marker of markers) {
+    const city = marker.city;
```

Comment:
since you only care about one field in marker, this could be a good use case for destructuring assignment.

```
for (const {city} of markers) {
  if (markerCounts[city] == null) // ...
}
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment